### ISSUE TITLE
NABULIB does not work with latest nightly (reccomended) z88dk

### ISSUE THREAD

**Issue by bocianu** [DJSures/NABU-LIB#9]
Hello World C example does not compile. 
Skeleton project also fails to compile. 

I have to download z88dk nightly version from may 2023 to be able to compile examples.

**Comment by linuxplayground** [comment]
+1 here.  
I am just encountering the same issue.  Can't compile my games.   I have yet to figure out what's changed though.

**Comment by linuxplayground** [comment]
Right.  I don't know WHY This fixes it, but it __SEEMS_ to work.  Maybe you can test on your end too.

in NABU-LIB.C, change the lines where it looks like `ld (INTERUPT_VECTOR_MAP_ADDRESS + ?),hl`

I don't think SDCC likes it when we do the addition inside the expression.  I tried hardcoding the actual address, and it started compiling ok.

INTERUPT_VECTOR_MAP_ADDRESS is defined as 0xFF00 in NABU-LIB.h

So I just searched for each occurrence of that variable in NABU-LIB.c and changed it to the hardcoded result of whatever expression was being used.

```
OLD:
ld (INTERUPT_VECTOR_MAP_ADDRESS + 6), hl;

NEW:
ld (0xFF06),hl
```

@DJSures  I know it's been a while since you looked at NABU stuff, but this library does need a bit of TLC.

All the function declarations that don't contain any arguments should be declared thusly:

```
void myFunc(void) {
   // 
}
```
IE: void inside the paratheses.

This fixes the warnings about missing prototypes.

**Comment by agmsmith** [comment]
NABU-LIB seems to be working now with the current Z88DK and related SDCC compiler.

**Comment by jefferoonie** [comment]
I get the same results as @bocianu.
I'm trying to compile on Linux. When I get back home, I will have to pull up the versions that I am using.

**Comment by agmsmith** [comment]
Which reminds me that I should get the latest Z88DK and compile it for Linux.  Have a look at my blog at https://web.ncf.ca/au829/WeekendReports/20240207/NthPongWarsBlog.html for the steps I went through to get the build environment working.

By the way, I have a hacked up NABU-LIB in GitHub to remove a lot of the warning messages from missing void function argument declarations (look in the NthPongCustomisations branch, that's the change in the December 28th 2025 commits, https://github.com/agmsmith/NABU-LIB/tree/NthPongCustomisations).  I've since hacked it up more for better missed frame detection (use a counter, not a boolean) and freeing up 2K of space (no full screen 80 column text buffer, redundant when using CP/M).

**Comment by jefferoonie** [comment]
Thank you for that information.  I pulled down your fork. The missing prototype errors went away.
I'll probably mess around with it in a few weeks.
